core/sdk-core/src/it/java/software/amazon/awssdk/core/http/AmazonHttpClientSslHandshakeTimeoutIntegrationTest.java

```diff
@@ -61,8 +64,7 @@ public void testSslHandshakeTimeout() {
                       .request(request)
                       .originalRequest(NoopTestRequest.builder().build())
                       .executionContext(executionContext(request))
-                      .errorResponseHandler(new NullErrorResponseHandler())
-                      .execute();
+                      .execute(combinedSyncResponseHandler(null, new NullErrorResponseHandler()));
```

Comment:
Is this common in any client?

Reply:
The execute() method on the client was internal and was only ever called by tests. I consider that an antipattern so I removed it, and made the test do a bit of extra heavy lifting instead.

Reply:
I meant having null on either both or one of the response handlers

Reply:
The null is handled by combinedSyncResponseHandler() which is a test helper method. There's nothing in the actual code that expects or handles a null for a handler (although there used to be through that test-only path).